### PR TITLE
Feat-integrate-apps-flyer-sdk

### PR DESCRIPTION
**Pull request by sultanmyrza**
┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201016280880500/1202416123287686) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by sultanmyrza** [comment]
First, I followed instructions from [Basic integration of the SDK (App Flyer)](https://github.com/AppsFlyerSDK/appsflyer-capacitor-plugin/blob/main/docs/BasicIntegration.md). ✅

File: src/app/shared/app-flyer/app-flyer.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import { AppFlyerService } from './app-flyer.service';
+
+describe('AppFlyerService', () => {
+  let service: AppFlyerService;
+  let platformIs: (platformName: string) => boolean;
+  let platformReadySpy: Promise<void>;
+  let platformSpy: Platform;
+
+  beforeEach(
+    waitForAsync(async () => {
```
Comment (sultanmyrza) [inline src/app/shared/app-flyer/app-flyer.service.spec.ts:13]: Currently this test fails with error

```
  Uncaught Error: Uncaught (in promise): TypeError: this.platform.is is not a function
  TypeError: this.platform.is is not a function
```

I found this stackoverflow [answer](https://stackoverflow.com/questions/66146320/ionic-typeerror-this-platform-is-is-not-a-function-when-running-unit-tests) and also saw similar code in [app.componen.spec.ts](https://github.com/numbersprotocol/capture-lite/blob/c485e6d662cc46490cf92a7d25ff5fe3b15fd1aa/src/app/app.component.spec.ts#L16) but after 1-2 hours of trying I couldn't fix this test. 

Need help to fix it from seniors 🥺.

Comment (bafu) [reply src/app/shared/app-flyer/app-flyer.service.spec.ts:13]: I did not try the code yet, and here are some thoughts

1. These materials might be helpful, but you likely have read them

https://jasmine.github.io/api/edge/jasmine.html#.createSpyObj

https://luardo.medium.com/testing-with-mocks-spies-using-createspyobj-in-angular-1e31bf7f6147

2. `platformReadySpy` is Object, and `platformIs` is function.

File: ios/App/App/Info.plist [hunk 2]
```diff
@@ -86,5 +86,7 @@
 	</array>
 	<key>UIViewControllerBasedStatusBarAppearance</key>
 	<true/>
+	<key>NSUserTrackingUsageDescription</key>
+	<string>Your data will be used to measure advertising efficiency</string>
```
Comment (bafu) [inline ios/App/App/Info.plist:90]: @sultanmyrza Is it possible to change the description to "We collect your data only for counting download number and will never share it with any 3rd party."

Comment (sultanmyrza) [reply ios/App/App/Info.plist]: @bafu, hope [fix(Info.plist): change NSUserTrackingUsageDescription](https://github.com/numbersprotocol/capture-lite/pull/1691/commits/815cb500700cded51b2a7afdbcfafff05fbbf7ec) commit resolves the issue

File: set-secret.js [hunk 3]
```diff
@@ -8,6 +8,7 @@ const envConfigFile = `
 export const BASE_URL = '${process.env.NUMBERS_STORAGE_BASE_URL}';
 export const TRUSTED_CLIENT_KEY = '${process.env.NUMBERS_STORAGE_TRUSTED_CLIENT_KEY}';
 export const BUBBLE_DB_URL = '${process.env.NUMBERS_BUBBLE_DB_URL}';
+export const APP_FLYER_DEV_KEY = '${process.env.APP_FLYER_DEV_KEY}'
```
Comment (bafu) [inline set-secret.js:11]: @sultanmyrza It should be `APPS_FLYER_DEV_KEY`.

File: src/app/shared/app-flyer/apps-flyer.service.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import { environment } from '../../../environments/environment';
+import { APP_FLYER_DEV_KEY } from '../dia-backend/secret';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AppsFlyerService {
+  private readonly afConfig: AFInit = {
+    appID: '1536388009', // AppStore Application ID. For iOS only.
+    devKey: APP_FLYER_DEV_KEY,
```
Comment (bafu) [inline src/app/shared/app-flyer/apps-flyer.service.ts:14]: @sultanmyrza It should be `APPS_FLYER_DEV_KEY`.

File: src/app/shared/app-flyer/apps-flyer.service.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    registerConversionListener: true,
+    registerOnAppOpenAttribution: false,
+    deepLinkTimeout: 4000, // default 3000 ms
+    useReceiptValidationSandbox: true, // iOS only
+    useUninstallSandbox: true, // iOS only
+  };
+
+  constructor(private readonly platform: Platform) {}
+
+  async initAppFlyerSDK() {
```
[the remark this answers is not in the document]
Comment (bafu) [reply src/app/shared/app-flyer/apps-flyer.service.ts:28]: Remember to update the codes calling this function at other places.

**Review by bafu — CHANGES_REQUESTED** [review]

**Pull request merged**